teos/src/tx_index.rs

```diff
@@ -281,15 +301,32 @@ mod tests {
 
         let cache: TxIndex<Locator, Transaction> = TxIndex::new(&last_n_blocks, height as u32);
 
-        assert_eq!(
-            cache.get_height(&first_block.block_hash()).unwrap(),
-            height - cache_size + 1
-        );
-        assert_eq!(cache.get_height(&last_block.block_hash()).unwrap(), height);
-        assert_eq!(
-            cache.get_height(&mid.block_hash()).unwrap(),
-            height - cache_size / 2
-        );
+        match first_block.deref() {
+            lightning_block_sync::BlockData::FullBlock(b) => {
+                assert_eq!(
+                    cache.get_height(&b.header.block_hash()).unwrap(),
+                    height - cache_size + 1
+                );
+            }
+            _ => panic!("Expected FullBlock"),
+        }
```

Comment:
and we can let `first_block` be:
```rust
let blocks = get_full_blocks(&last_n_blocks); // kinda redundant clone just for the first block, but that's OK since these are tests.
let first_block = blocks.first().unwrap();
...
...
assert_eq!(cache.get_height(&b.header.block_hash()).unwrap(), height - cache_size + 1);
```